tests/lib.rs

```diff
@@ -1 +1,3 @@
+#![allow(clippy::cognitive_complexity)]
```

Comment:
Clippy doesn't like some of our longer test functions, so I just blanket whitelisted all of them in the tests